remix-analyzer/src/solidity-analyzer/modules/selfdestruct.js

```diff
@@ -2,25 +2,51 @@ var name = 'Selfdestruct: '
 var desc = 'Be aware of caller contracts.'
 var categories = require('./categories')
 var common = require('./staticAnalysisCommon')
+var AbstractAst = require('./abstractAstView')
 
 function selfdestruct () {
-  this.relevantNodes = []
-}
+  this.abstractAst = new AbstractAst()
+
+  this.visit = this.abstractAst.build_visit(
+    (node) => common.isStatement(node) ||
+              common.isSelfdestructCall(node)
+  )
 
-selfdestruct.prototype.visit = function (node) {
-  if (common.isSelfdestructCall(node)) {
-    this.relevantNodes.push(node)
-  }
+  this.report = this.abstractAst.build_report(report)
 }
 
-selfdestruct.prototype.report = function () {
-  return this.relevantNodes.map(function (item, i) {
-    return {
-      warning: 'Use of selfdestruct: can block calling contracts unexpectedly. Be especially careful if this contract is planned to be used by other contracts (i.e. library contracts, interactions). Selfdestruction of the callee contract can leave callers in an inoperable state.',
-      location: item.src,
-      more: 'https://paritytech.io/blog/security-alert.html'
-    }
+selfdestruct.prototype.visit = function () { throw new Error('constantFunctions.js no visit function set upon construction') }
```

Comment:
out of curiosity why this change?

Reply:
What do you mean exactly?

Reply:
this PR add a default behavior to the `prototype`, and override it line 10

Reply:
That is actually a good question. I use this pattern whenever i use the abstractAstView in modules. I can't remember the rational i had when i wrote that. Maybe just as safeguard for myself.